src/backend/distributed/executor/multi_utility.c

```diff
@@ -312,12 +313,15 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 
 	if (IsA(parsetree, CopyStmt))
 	{
-		/* copy parse tree since we might scribble on it to fix the schema name */
-		parsetree = copyObject(parsetree);
+		MemoryContext planCtx = GetMemoryChunkContext(parsetree);
+		MemoryContext previousContext = MemoryContextSwitchTo(planCtx);
 
+		parsetree = copyObject(parsetree);
 		parsetree = ProcessCopyStmt((CopyStmt *) parsetree, completionTag,
```

Comment:
Alternately, you could refactor all of `ProcessCopyStmt` to ensure it never mutates the `parsetree` it is handed. Instead, you'd declare variables like `newSchemaName` or `newFileName` or `newQuery` and use them to build a new `CopyStatement` at the very bottom of the method (in the right memory context), and return _that_. This is probably both cleanest and most performant in the long run, but I wouldn't mind if you go with the double-copy + a `TODO` to get unblocked.